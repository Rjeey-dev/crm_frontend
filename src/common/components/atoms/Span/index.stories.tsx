import React from 'react';
import { storiesOf } from '@storybook/react';
import Span from './index';

storiesOf('Span', module)
    .add('With text', () => (
        <Span classes='test' text='Some text'/>
    ))
    .add('Without text', () => (
        <Span classes='test'/>
    ))
    .add('Without classes', () => (
        <Span text='Some text'/>
    ))
    .add('Without classes and text', () => (
        <Span/>
    ));