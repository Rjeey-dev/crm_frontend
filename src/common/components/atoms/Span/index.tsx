import React, {ReactNode, ReactNodeArray} from "react";

interface IProps {
    classes?: string,
    id?: string,
    children?: ReactNodeArray | ReactNode,
    attributes?: object,
    onClick?: (event: React.FormEvent<HTMLSpanElement>) => void
}

function Span(props: IProps) {
    return <span onClick={props.onClick} className={props.classes} {...props.attributes} id={props.id}>{props.children}</span>;
}

export default Span;
