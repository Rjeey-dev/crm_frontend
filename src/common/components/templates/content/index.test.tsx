import {shallow} from "enzyme";
import React from 'react';

import Span from '../../atoms/Span/index';
import ContentTemplate from './index';

it("Renders <ContentTemplate/> with correct properties", () => {
    const output = shallow(
        <ContentTemplate>
            <Span/>
        </ContentTemplate>
    ).html();

    expect(output).toMatchSnapshot();
});
