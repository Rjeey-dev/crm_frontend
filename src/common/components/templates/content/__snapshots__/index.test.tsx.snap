// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <ContentTemplate/> with correct properties 1`] = `"<div class=\\"container-fluid page-body-wrapper\\"><span></span></div>"`;
